src/test/java/org/jboss/logmanager/handlers/PeriodicRotatingFileHandlerTests.java

```diff
@@ -0,0 +1,146 @@
+
+    private final String logFile = BASE_LOG_DIR.toString() + "/" + FILENAME;
+
+    private final SimpleDateFormat rotateFormatter = new SimpleDateFormat(".yyyy-MM-dd");
+    private PeriodicRotatingFileHandler handler;
+
+    @Before
+    public void createHandler() throws FileNotFoundException {
+        // Create the handler
+        File logFile = new File(this.logFile);
+        handler = new PeriodicRotatingFileHandler(logFile, rotateFormatter.toPattern(), false);
+        handler.setFormatter(FORMATTER);
+    }
+
+    @After
+    public void closeHandler() {
+        handler.close();
+        handler = null;
+    }
+
+    @Test
+    public void testRotate() throws Exception {
+        final Calendar cal = Calendar.getInstance();
+        final String rotatedPath = BASE_LOG_DIR.toString() + "/" + FILENAME + rotateFormatter.format(cal.getTime());
+        File rotatedFile = new File(rotatedPath);
+        testRotate(cal, rotatedFile);
+    }
+
+    @Test
+    public void testOverwriteRotate() throws Exception {
+        final Calendar cal = Calendar.getInstance();
+        final String rotatedPath = BASE_LOG_DIR.toString() + "/" + FILENAME + rotateFormatter.format(cal.getTime());
+        
+        // Create the rotated file to ensure at some point it gets overwritten
+        File rotatedFile = new File(rotatedPath);
+        
+        if (rotatedFile.exists())
+            rotatedFile.delete();
+        
+        try {
+            FileWriter fwriter = new FileWriter(rotatedFile.getAbsolutePath());
+	    BufferedWriter  writer = new BufferedWriter(fwriter);
+            writer.write("Adding data to the file");
+            writer.close(); 
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        testRotate(cal, rotatedFile);
+    }
+
+    private void testRotate(final Calendar cal, final File rotatedFile) throws Exception {
+        final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+        final int currentDay = cal.get(Calendar.DAY_OF_MONTH);
+        final int nextDay = currentDay + 1;
+        File logFile = new File(this.logFile);
+
+        final String currentDate = sdf.format(cal.getTime());
+
+        // Create a log message to be logged
+        ExtLogRecord record = createLogRecord(Level.INFO, "Date: %s", currentDate);
+        handler.publish(record);
+
+        Assert.assertTrue("File '" + logFile.getName() + "' does not exist", logFile.exists());
+
+        // Read the contents of the log file and ensure there's only one line and that like
+        List<String> lines = readAllLines(logFile);
+        Assert.assertEquals("More than 1 line found", 1, lines.size());
+        Assert.assertTrue("Expected the line to contain the date: " + currentDate, lines.get(0).contains(currentDate));
+
+        // Create a new record, increment the day by one and validate
+        cal.add(Calendar.DAY_OF_MONTH, nextDay);
```

Comment:
I had this same thing in my original test, but it was kind of wrong. It should just be `cal.add(Calendar.DAY_OF_MONTH, 1);`. This makes the `nextDay` and `currentDay` variables go away too.